.github/workflows/bencher.yml

```diff
@@ -42,22 +46,76 @@ env:
   BENCHER_PROJECT: ${{ vars.BENCHER_PROJECT || 'servo' }}
 
 jobs:
-  bencher:
-    name: Bencher (${{ inputs.target }})
-    # This needs to be kept in sync with the `--testbed` argument sent to bencher.
+  # Runs the underlying job (“workload”) on a self-hosted runner if available,
+  # with the help of a `runner-select` job and a `runner-timeout` job.
+  runner-select:
     runs-on: ubuntu-22.04
+    outputs:
+      unique-id: ${{ steps.select.outputs.unique-id }}
+      selected-runner-label: ${{ steps.select.outputs.selected-runner-label }}
+      selected-runner-host: ${{ steps.select.outputs.selected-runner-host }}
+      is-self-hosted: ${{ steps.select.outputs.is-self-hosted }}
     steps:
-      - uses: actions/checkout@v4
-        if: github.event_name != 'pull_request_target'
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: '.github'
+      - name: Runner select
+        id: select
+        uses: ./.github/actions/runner-select
+        with:
+          monitor-api-token: ${{ secrets.SERVO_CI_MONITOR_API_TOKEN }}
+          github-hosted-runner-label: ubuntu-22.04
```

Comment:
Is there a usecase for falling back to github hosted runners for benchmarks?

Reply:
in general, we try to make our usage of self-hosted runners gracefully degrade to github-hosted runners, because self-hosted runners are provided on a best-effort basis. and for bencher jobs, there are cases where we only measure binary size, so we use `force-github-hosted-runner` to force those cases to use github-hosted runners.